### PR TITLE
Game OSD: Add "Advanced settings" button to settings list

### PR DESCRIPTION
**Pull request by garbear**
This adds an "advanced settings" button to the in-game OSD. It links to the add-on settings for the active game client.

## Motivation and Context
Requested a long time ago.

## How Has This Been Tested?
Tested on Windows 10 with several game clients.

## Screenshots (if appropriate):
game advanced settings

## Types of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves existing functionality)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: xbmc/cores/RetroPlayer/guibridge/GUIGameRenderManager.cpp [hunk 1]
```diff
@@ -100,6 +116,11 @@ std::shared_ptr<CGUIGameVideoHandle> CGUIGameRenderManager::RegisterDialog(GAME:
   return std::make_shared<CGUIGameVideoHandle>(*this);
 }
 
+std::shared_ptr<CGUIGameSettingsHandle> CGUIGameRenderManager::RegisterDialog(GAME::CDialogGameAdvancedSettings &dialog)
+{
+  return std::make_shared<CGUIGameSettingsHandle>(*this);
+}
+
```
Comment (a1rwulf) [inline xbmc/cores/RetroPlayer/guibridge/GUIGameRenderManager.cpp]: Where is the dialog param used or why do you need it?

Comment (garbear) [reply xbmc/cores/RetroPlayer/guibridge/GUIGameRenderManager.cpp]: A parameter is needed for the game window and game controls, and I just copied the pattern for game settings. It's not needed so I'll remove it.

**Comment by garbear** [comment]
jenkins build this please

**Comment by a1rwulf** [comment]
Thx @garbear

**Pull request merged**